### ISSUE TITLE
Allow disabling of alias generation

### ISSUE THREAD

**Issue by RevCBH** [dillonkearns/elm-graphql#106]
This is a feature request.

In trying to use this library with AWS AppSync, I determined that the subscription feature of that service fails to trigger due to the auto-generated aliases on mutation requests.

For example, this generated mutation performs the update, but fails to trigger a subscription event:

```
mutation { 
  startShow3832717217: startShow(date: "2019-02-02", time: "02:00 pm") { 
    showState3523593770: showState 
    comedians2741155849: comedians 
    time3832528868: time 
    date3832528868: date 
  } 
}
```

Whereas a comparable, non-aliased mutation does both.

As such, I'd be keen to disable the aliasing. Happy to take a crack at an implementation, but wanted to see if you had thoughts on the feature before I go whole hog on it.

**Comment by dillonkearns** [comment]
Hello @RevCBH, thanks for opening the thread. This is a pretty tricky issue because this library builds up aliases in order to guarantee that there are no collisions, and it decides this aliases without any context except the current field's type and arguments.

Aliasing individual fields wouldn't be necessary if it weren't for this one arbitrary rule that was introduced in the GraphQL spec. See [Counter Example No 112](https://facebook.github.io/graphql/draft/#example-54e3d) from the GraphQL Spec. You can read all the details where we decided on this approach here: https://github.com/dillonkearns/elm-graphql/issues/95. I wish it wasn't necessary to account for this corner case, but because of this we need to use different aliases in the case where two fields have the same name but different types in an `on type {...}` selection.

I can't think of a way to get around this except for introducing context (but even in that case you would only be able to remove aliases for most cases, not all, see the exception described in the issue above). You can read about why I do the field hashes without any context here, it's allowed for a lot of nice feature development: https://medium.com/@dillonkearns/how-elm-guides-towards-simplicity-3d34685dc33c.

It would be a shame to introduce a lot of complexity, or special features just to work around some bug cases in some services. I've heard about a bug related to field aliases in `graphql-yoga` as well, which is unfortunate.

But aliasing fields is a pretty basic element of the GraphQL Spec, so I would expect services to fix bugs related to this. Have you reported this bug with the AppSync team? Let's start there before we discuss adding complication to workaround bugs. It's not ideal either way. In the meantime, if need be, you can clone this repo into `vendor` and remove aliasing for fields.

Thanks!

**Comment by RevCBH** [comment]
Thanks for the in depth reply, that makes a lot of sense! Will definitely ping the AppSync team, but they appear to be aware of the issue and uninterested in addressing it.

I'm looking at workarounds right now that don't require forking the library. If I get one working I'll do a quick writeup for anyone else that might be interested.

**Comment by dillonkearns** [comment]
Sounds good, thanks for the update @RevCBH. Could you please link me to any relevant issues that you are aware of right now? And keep me posted on any new issues you create. I'd love to see what their response is. Thank you!

**Comment by RevCBH** [comment]
Yes, will do. I'll try to find the forum post again, it took a bit of searching last time.

For posterity, here's the hacked up version of `Graphql.Http` I ended up with to remove the top-level aliases on mutation requests that should trigger subscriptions: https://gist.github.com/RevCBH/78ff13fc6e208abdee7b7269209a8ec3
